drivers/net/bnxt/bnxt_cpr.c

```diff
@@ -97,7 +99,19 @@ void bnxt_handle_fwd_req(struct bnxt *bp, struct cmpl_base *cmpl)
 		goto reject;
 	}
 
-	if (bnxt_rcv_msg_from_vf(bp, fw_vf_id - bp->pf.first_vf_id, fwd_cmd) == true) {
+	if (bnxt_rcv_msg_from_vf(bp, vf_id, fwd_cmd) == true) {
+		/*
+		 * In older firmware versions, the MAC had to be all zeros for the
+		 * VF to set it's MAC via hwrm_func_vf_cfg.  Set to all zeros if it's
+		 * being configured and has been approved by the app
+		 */
+		if (fwd_cmd->req_type == HWRM_FUNC_VF_CFG) {
+			struct hwrm_func_vf_cfg_input *vfc = (void *)fwd_cmd;
+
+			if (vfc->enables & HWRM_FUNC_VF_CFG_INPUT_ENABLES_DFLT_MAC_ADDR) {
+				bnxt_hwrm_func_vf_mac(bp, vf_id, (const uint8_t *)"\x00\x00\x00\x00\x00");
```

Comment:
So we set force the current MAC to zero just before the VF's new MAC is forwarded and approved?